sql/catalyst/src/main/java/org/apache/spark/sql/connector/expressions/aggregate/UserDefinedAggregateFunc.java

```diff
@@ -50,4 +53,18 @@ public UserDefinedAggregateFunc(
 
   @Override
   public Expression[] children() { return children; }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    UserDefinedAggregateFunc that = (UserDefinedAggregateFunc) o;
+    return Objects.equals(name, that.name) && Objects.equals(canonicalName, that.canonicalName) &&
+      Arrays.equals(children, that.children);
```

Comment:
(added `isDistinct == that.isDistinct`)
```suggestion
    return isDistinct == that.isDistinct && Objects.equals(name, that.name) &&
      Objects.equals(canonicalName, that.canonicalName) && Arrays.equals(children, that.children);
```